### PR TITLE
feat: Support TLS connections

### PR DESCRIPTION
**Pull request by EmilHernvall**
TLS is supported by redis-rs, but the information was not being correctly plumbed through. This passes the information on correctly and allows for successful establishments of connections, as tested with an AWS Elasticache cluster.

(Hi Markus! How's the new job?)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -460,13 +466,22 @@ where
                         Some(pw) => format!("redis://:{}@{}:{}", pw, host, port),
                         None => format!("redis://{}:{}", host, port),
                     },
+                    ConnectionAddr::TcpTls { ref host, port, insecure } => match &info.redis.password {
+                        Some(pw) if insecure => format!("rediss://:{}@{}:{}/#insecure", pw, host, port),
```
Comment (Marwes) [inline src/lib.rs:470]:
```suggestion
                        Some(pw) => format!("rediss://:{}@{}:{}{}", pw, host, port, if insecure { "/#insecure" } else  { "" }),
```
Would remove two branches branches

File: src/lib.rs [hunk 2]
```diff
@@ -434,6 +435,10 @@ where
     C: ConnectionLike + Connect + Clone + Send + Sync + 'static,
 {
     async fn new(initial_nodes: &[ConnectionInfo], retries: Option<u32>) -> RedisResult<Self> {
+        let tls = initial_nodes.iter().all(|c| match c.addr {
+            ConnectionAddr::TcpTls { .. } => true,
```
Comment (Marwes) [inline src/lib.rs:439]: This should probably check that all nodes use tls or none of them does, else it will break during reconnections.

**Review by Marwes — APPROVED** [review]
LGTM. Would be nice with some TLS tests but that would be easier to do if this was merged into redis-rs proper.

**Comment by Marwes** [comment]
> (Hi Markus! How's the new job?)

New work is great so far! Actually getting to leverage the knowledge I gained from gluon for pay!

**Comment by BratSinot** [comment]
Hi @Marwes can you merge these changes please?
Also, you forgot to enable `tls` feature in `redis` crate.

**Comment by EmilHernvall** [comment]
Working on a follow up patch to address Marwes' comments. I'll try to get that out today.

**Comment by mpetri** [comment]
Any updates towards the follow up patch @EmilHernvall ?

**Comment by tasn** [comment]
Would also love to have this in. 🙏🏻

**Comment by tasn** [comment]
Yay, thanks!

**Pull request merged**